lib/ts/recipe/thirdpartyemailpassword/utils.ts

```diff
@@ -0,0 +1,411 @@
+}
+
+function validateAndNormaliseSignUpConfig(
+    recipeInstance: Recipe,
+    appInfo: NormalisedAppinfo,
+    config?: TypeInputSignUp
+): TypeNormalisedInputSignUp {
+    let disableDefaultImplementation =
+        config === undefined || config.disableDefaultImplementation === undefined
+            ? false
+            : config.disableDefaultImplementation;
+
+    let formFields: NormalisedFormField[] = [];
+    if (config !== undefined && config.formFields !== undefined) {
+        config.formFields.forEach((field) => {
+            if (field.id === FORM_FIELD_PASSWORD_ID) {
+                formFields.push({
+                    id: field.id,
+                    validate: field.validate === undefined ? defaultPasswordValidator : field.validate,
+                    optional: false,
+                });
+            } else if (field.id === FORM_FIELD_EMAIL_ID) {
+                formFields.push({
+                    id: field.id,
+                    validate: field.validate === undefined ? defaultEmailValidator : field.validate,
+                    optional: false,
+                });
+            } else {
+                formFields.push({
+                    id: field.id,
+                    validate: field.validate === undefined ? defaultValidator : field.validate,
+                    optional: field.optional === undefined ? false : field.optional,
+                });
+            }
+        });
+    }
+    if (formFields.filter((field) => field.id === FORM_FIELD_PASSWORD_ID).length === 0) {
+        // no password field give by user
+        formFields.push({
+            id: FORM_FIELD_PASSWORD_ID,
+            validate: defaultPasswordValidator,
+            optional: false,
+        });
+    }
+    if (formFields.filter((field) => field.id === FORM_FIELD_EMAIL_ID).length === 0) {
+        // no email field give by user
+        formFields.push({
+            id: FORM_FIELD_EMAIL_ID,
+            validate: defaultEmailValidator,
+            optional: false,
+        });
+    }
+
+    let handlePostSignUp: (user: User, context: TypeContextEmailPassowrd | TypeContextThirdParty) => Promise<void> =
+        config === undefined || config.handlePostSignUp === undefined
+            ? defaultHandlePostSignUp
+            : config.handlePostSignUp;
+
+    return {
+        disableDefaultImplementation,
+        formFields,
+        handlePostSignUp,
+    };
+}
+
+function validateAndNormaliseSessionFeatureConfig(
+    recipeInstance: Recipe,
+    appInfo: NormalisedAppinfo,
+    config?: TypeInputSessionFeature
+): TypeNormalisedInputSessionFeature {
+    let setJwtPayload: (
+        user: User,
+        context: TypeContextEmailPassowrd | TypeContextThirdParty,
+        action: "signin" | "signup"
+    ) => Promise<{ [key: string]: any }> =
+        config === undefined || config.setJwtPayload === undefined
+            ? defaultSetJwtPayloadForSession
+            : config.setJwtPayload;
+
+    let setSessionData: (
```

Comment:
I don't think you need to declare types here.. TS infers it based on the return type.

Reply:
@bhumilsarvaiya this applies to all the places in the utils.ts file.. try and remove as many of these unnecessary type declerations.

Reply:
done